lib/form/js/repeatable.js

```diff
@@ -0,0 +1,245 @@
+        newElement.type = 'button';
+        newElement.innerHTML = 'Delete';
+        newElement.id = input + '-del';
+        newElement.setAttribute('class', 'delete-button')
+        newElement.addEventListener("click", deleteHandler);
+        container.appendChild(newElement);
+        container.id = container.id + '--' + (allRepeatables[input].length-1);
+        var childNodes = container.querySelectorAll('*')
+        childNodes.forEach( function(child) {
+          if (child.id != "") {
+            child.id = child.id + '--' + (allRepeatables[input].length-1);
+            var name = child.getAttribute('name');
+            if (name) {
+              child.setAttribute('name', name +  '--' + (allRepeatables[input].length-1));
+            }
+          }
+          else if (child.htmlFor && child.htmlFor != "") {
+            child.htmlFor = child.htmlFor + '--' + (allRepeatables[input].length-1);
+          }
+        });
+      }
+
+      // deep clones the existing instance
+      var newContainer = container.cloneNode(true);
+      var idMap = {};
+
+      newContainer.id = newContainer.id.split('--')[0] + '--' + allRepeatables[input].length;
+      var childNodes = newContainer.querySelectorAll('*');
+      childNodes.forEach( function (child) {
+        if (child.id != ""){
+          id = splitPop(child.id, '--');
+          var name = child.getAttribute('name');
+          if (name) {
+            name = splitPop(name, '--');
+            child.setAttribute('name', name +  '--' + allRepeatables[input].length);
+          }
+
+          if (idMap.hasOwnProperty(id)){
+            child.id = idMap[id];
+          }
+          else {
+            var newId = id + '--' + allRepeatables[input].length;
+            idMap[id] = newId;
+            child.id = newId;
+          }
+
+          if (child.type == 'checkbox' || child.type == 'radio') {
+            child.checked = false;
+          }
+          else if (child.selectedIndex != null) {
+            child.selectedIndex = 0;
+          }
+          else if (child.id.indexOf(idDel) < 0){
+            child.value = '';
+          }
+
+          if (child.id.indexOf(idDel) > -1) {
+            child.addEventListener("click", deleteHandler);
+          }
+          if (!focusChild) {
+            focusChild = child;
+          }
+        }
+        else if (child.htmlFor && child.htmlFor != "") {
+          var htmlFor = splitPop(child.htmlFor, '--');
+          if (idMap.hasOwnProperty(htmlFor)){
+            child.htmlFor = idMap[htmlFor];
+          }
+          else {
+            var newHtmlFor = htmlFor + '--' + allRepeatables[input].length;
+            idMap[htmlFor] = newHtmlFor;
+            child.htmlFor = newHtmlFor;
+          }
+        }
+      });
+
+      // Increments length of attribute
+      allRepeatables[input].length++;
+      var containerSet = container.parentNode.querySelectorAll('div.form-repeatable');
```

Comment:
Is there an instance where `.form--repeatable` isn't on a div? We don't need to over-qualify this selector